test/rules/no-optional-parameter-in-callback/test.ts.lint

```diff
@@ -0,0 +1,86 @@
+// good
+declare function f1(a: number);
+
+declare function f2(a?: number, b: (c: string, d: number) => void);
+
+interface i1 {
+    f1();
+    f2(a: symbol);
+    f3(a: number, b?: (c: string, d: number) => void);
+}
+
+class c1 {
+    ab(): number;
+	ab(x?: number, y?: number): number {
+		return 1;
+    }
+}
+
+function f3() {
+    function f4(x?: number, y?: (k: string) => void): number {
+        return 1;
+    }
+}
+
+module MyModule {
+    export class ClassA {
+        id: string;
+        ab(x?: number, y?: number): number {
+		    return 1;
+        }
+    }
+
+    export class ClassB {
+        id: string;
+    }
+}
+
+
+
+
+// bad
+declare function bar(a: (b: number, k?: string));
+                                    ~~~~~~~~~~  [error]
+
+function f3() {
+    function f4(x?: number, y?: (k?: string) => void): number {
+                                 ~~~~~~~~~~  [error]
+        return 1;
+    }
+}
+
+var v1 = {
+    foo(x: (y?: number): void {
```

Comment:
This line has a syntax error on it. It should be `    foo(x: (y?: number) => void): void {`